internal/logger/http_logger.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 Canonical.
+
+package logger
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+)
+
+// HTTPLogFormatter is an implementation of chimiddleware.LogFormatter. It formats logs for http requests.
+type HTTPLogFormatter struct{}
+
+// httpLogEntry is an implementation of chimiddleware.LogEntry.
+type httpLogEntry struct {
+	*HTTPLogFormatter
+	request *http.Request
+}
+
+// Write is called when the request handler has finished.
+func (l *httpLogEntry) Write(status, bytes int, header http.Header, elapsed time.Duration, extra interface{}) {
+	fields := make([]zap.Field, 0)
+
+	if status != 0 {
```

Comment:
I don't think this would be debug only, might we want it on all the time but only log specific status codes, and then perhaps if its anything other than a erroneous code, then default to debug?